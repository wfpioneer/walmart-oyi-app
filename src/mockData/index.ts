
import itemDetails from './getItemDetails';


export const getMockItemDetails = (upc: string) => {
  // @ts-ignore
  return itemDetails[upc] || {};
};
