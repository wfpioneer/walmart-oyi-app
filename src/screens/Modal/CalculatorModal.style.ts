import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    alignItems: 'center'
  },
  buttonContainer: {
    alignContent: 'space-around',
    flexDirection: 'row',
    padding: 10,
    height: 60
  },
  closeBtn: {
    width: '100%'
  }
});

export default styles;
