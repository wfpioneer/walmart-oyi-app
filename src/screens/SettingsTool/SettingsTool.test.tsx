import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { NavigationProp } from '@react-navigation/native';
import Toast from 'react-native-toast-message';
import { fireEvent, render } from '@testing-library/react-native';
import {
  CollapsibleCard, SettingsToolScreen, featureCard, getConfigAndFluffyFeaturesApiHook, printerCard
} from './SettingsTool';
import { mockPrinterList } from '../../mockData/mockPrinterList';
import { Configurations } from '../../models/User';
import { mockConfig } from '../../mockData/mockConfig';
import mockUser from '../../mockData/mockUser';
import { strings } from '../../locales';
import { SNACKBAR_TIMEOUT } from '../../utils/global';
import { hideActivityModal, showActivityModal } from '../../state/actions/Modal';

jest.mock('react-native-vector-icons/MaterialIcons', () => 'Icon');

jest.mock('../../state/actions/Modal', () => ({
  showActivityModal: jest.fn(),
  hideActivityModal: jest.fn()
}));

describe('SettingsToolScreen', () => {
  const testConfigs: Configurations = {
    ...mockConfig,
    locationManagement: true,
    locationManagementEdit: true,
    settingsTool: true,
    palletManagement: true,
    binning: true,
    picking: true,
    printingUpdate: true
  };
  const navigationProp: NavigationProp<any> = {
    addListener: jest.fn(),
    canGoBack: jest.fn(),
    getParent: jest.fn(),
    getId: jest.fn(),
    getState: jest.fn(),
    dispatch: jest.fn(),
    goBack: jest.fn(),
    isFocused: jest.fn(() => true),
    removeListener: jest.fn(),
    reset: jest.fn(),
    setOptions: jest.fn(),
    setParams: jest.fn(),
    navigate: jest.fn()
  };

  const defaultAsyncState = {
    isWaiting: false,
    value: null,
    error: null,
    result: null
  };
  describe('Tests rendering the SettingsToolScreen component', () => {
    it('Test renders the default SettingsToolScreen ', () => {
      const renderer = ShallowRenderer.createRenderer();
      renderer.render(
        <SettingsToolScreen
          printerOpen={true}
          togglePrinterList={jest.fn()}
          featuresOpen={true}
          toggleFeaturesList={jest.fn()}
          priceLabelPrinter={null}
          locationLabelPrinter={null}
          palletLabelPrinter={null}
          userFeatures={[]}
          userConfigs={testConfigs}
          dispatch={jest.fn()}
          navigation={navigationProp}
          getClubConfigApiState={defaultAsyncState}
          getFluffyApiState={defaultAsyncState}
          useEffectHook={jest.fn}
          user={mockUser}
          trackEventCall={jest.fn()}
        />
      );
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });

    it('Test renders the SettingsToolScreen with Selected Printers', () => {
      const renderer = ShallowRenderer.createRenderer();
      renderer.render(
        <SettingsToolScreen
          printerOpen={true}
          togglePrinterList={jest.fn()}
          featuresOpen={true}
          toggleFeaturesList={jest.fn()}
          priceLabelPrinter={mockPrinterList[0]}
          locationLabelPrinter={mockPrinterList[1]}
          palletLabelPrinter={mockPrinterList[2]}
          userFeatures={[]}
          userConfigs={testConfigs}
          dispatch={jest.fn()}
          navigation={navigationProp}
          getClubConfigApiState={defaultAsyncState}
          getFluffyApiState={defaultAsyncState}
          useEffectHook={jest.fn}
          user={mockUser}
          trackEventCall={jest.fn()}
        />
      );
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });

    it('Test renders the SettingsToolScreen with the Printer List collapsed', () => {
      const renderer = ShallowRenderer.createRenderer();
      renderer.render(
        <SettingsToolScreen
          printerOpen={false}
          togglePrinterList={jest.fn()}
          featuresOpen={true}
          toggleFeaturesList={jest.fn()}
          priceLabelPrinter={null}
          locationLabelPrinter={null}
          palletLabelPrinter={null}
          userFeatures={[]}
          userConfigs={testConfigs}
          dispatch={jest.fn()}
          navigation={navigationProp}
          getClubConfigApiState={defaultAsyncState}
          getFluffyApiState={defaultAsyncState}
          useEffectHook={jest.fn}
          user={mockUser}
          trackEventCall={jest.fn()}
        />
      );
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });

    it('Test renders the SettingsToolScreen with the Feature List collapsed', () => {
      const renderer = ShallowRenderer.createRenderer();
      renderer.render(
        <SettingsToolScreen
          printerOpen={true}
          togglePrinterList={jest.fn()}
          featuresOpen={false}
          toggleFeaturesList={jest.fn()}
          priceLabelPrinter={null}
          locationLabelPrinter={null}
          palletLabelPrinter={null}
          userFeatures={[]}
          userConfigs={testConfigs}
          dispatch={jest.fn()}
          navigation={navigationProp}
          getClubConfigApiState={defaultAsyncState}
          getFluffyApiState={defaultAsyncState}
          useEffectHook={jest.fn}
          user={mockUser}
          trackEventCall={jest.fn()}
        />
      );
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
    it('Tests functionality of update button in settings tool screen', () => {
      const mockDispatch = jest.fn();
      const { getByTestId } = render(
        <SettingsToolScreen
          printerOpen={true}
          togglePrinterList={jest.fn()}
          featuresOpen={false}
          toggleFeaturesList={jest.fn()}
          priceLabelPrinter={null}
          locationLabelPrinter={null}
          palletLabelPrinter={null}
          userFeatures={[]}
          userConfigs={testConfigs}
          dispatch={mockDispatch}
          navigation={navigationProp}
          getClubConfigApiState={defaultAsyncState}
          getFluffyApiState={defaultAsyncState}
          useEffectHook={jest.fn}
          user={mockUser}
          trackEventCall={jest.fn()}
        />
      );
      const updateButton = getByTestId('updateButton');
      fireEvent.press(updateButton);
      expect(mockDispatch).toHaveBeenCalledTimes(2);
    });
  });

  describe('Tests rendering featureCard component', () => {
    const renderer = ShallowRenderer.createRenderer();
    it('featureCard function renders "Enabled" translation if isEnabled is set to True', () => {
      renderer.render(
        featureCard('manager approval', true)
      );
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });

    it('featureCard function renders "Disabled" translation if isEnabled is set to False', () => {
      renderer.render(
        featureCard('manager approval', false)
      );
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
  });
  describe('Tests rendering printerCard component', () => {
    const renderer = ShallowRenderer.createRenderer();

    it('printerCard functions renders Printer Name if Printer is not null', () => {
      renderer.render(printerCard('Price Sign Printer', mockPrinterList[0], jest.fn()));
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });

    it('printerCard functions renders "Not Assigned" translation if Printer is null', () => {
      renderer.render(printerCard('Price Sign Printer', null, jest.fn()));
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
  });

  describe('Tests rendering CollapsibleCard component', () => {
    const renderer = ShallowRenderer.createRenderer();

    it('CollapsibleCard functions renders down-arrow  if isOpened is false', () => {
      renderer.render(
        <CollapsibleCard
          title=""
          isOpened={false}
          toggleIsOpened={jest.fn()}
          trackEventCall={jest.fn()}
          trackEventInfo={{ action: 'test' }}
        />
      );
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
    it('CollapsibleCard functions renders up-arrow  if isOpened is true', () => {
      renderer.render(
        <CollapsibleCard
          title=""
          isOpened={true}
          toggleIsOpened={jest.fn()}
          trackEventCall={jest.fn()}
          trackEventInfo={{ action: 'test' }}
        />
      );
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
  });
  describe('test externalized functions', () => {
    const mockDispatch = jest.fn();
    const successApi = {
      ...defaultAsyncState,
      result: {
        data: 'test',
        status: 200
      }
    };
    const loadingApiState = {
      ...defaultAsyncState,
      isWaiting: true
    };
    const errorApi = {
      ...defaultAsyncState,
      error: 'testError'
    };
    afterEach(() => {
      jest.clearAllMocks();
    });
    it('test getConfigAndFluffyFeaturesApiHook with success response', () => {
      getConfigAndFluffyFeaturesApiHook(successApi, successApi, navigationProp, mockDispatch);
      expect(mockDispatch).toHaveBeenCalledTimes(5);
      expect(hideActivityModal).toBeCalledTimes(1);
      expect(Toast.show).toHaveBeenCalledWith({
        type: 'success',
        text1: strings('SETTINGS.FEATURE_UPDATE_SUCCESS'),
        visibilityTime: SNACKBAR_TIMEOUT,
        position: 'bottom'
      });
    });
    it('test getConfigAndFluffyFeaturesApiHook with error response', () => {
      getConfigAndFluffyFeaturesApiHook(errorApi, errorApi, navigationProp, mockDispatch);
      expect(mockDispatch).toHaveBeenCalledTimes(3);
      expect(hideActivityModal).toBeCalledTimes(1);
      expect(Toast.show).toHaveBeenCalledWith({
        type: 'error',
        text1: strings('SETTINGS.FEATURE_UPDATE_FAILURE'),
        visibilityTime: SNACKBAR_TIMEOUT,
        position: 'bottom'
      });
    });
    it('test getConfigAndFluffyFeaturesApiHook during API request', () => {
      getConfigAndFluffyFeaturesApiHook(loadingApiState, loadingApiState, navigationProp, mockDispatch);
      expect(mockDispatch).toHaveBeenCalledTimes(1);
      expect(showActivityModal).toBeCalledTimes(1);
      expect(Toast.show).not.toBeCalled();
    });
  });
});
