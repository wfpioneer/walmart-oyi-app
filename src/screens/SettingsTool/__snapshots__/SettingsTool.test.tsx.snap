// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsToolScreen Tests rendering CollapsibleCard component CollapsibleCard functions renders down-arrow  if isOpened is false 1`] = `
<React.Fragment>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      
    </Text>
  </View>
  <TouchableOpacity
    onPress={[Function]}
    style={
      Object {
        "flex": 0.1,
        "paddingLeft": 5,
      }
    }
  >
    <Icon
      color="#616161"
      name="keyboard-arrow-down"
      size={25}
    />
  </TouchableOpacity>
</React.Fragment>
`;

exports[`SettingsToolScreen Tests rendering CollapsibleCard component CollapsibleCard functions renders up-arrow  if isOpened is true 1`] = `
<React.Fragment>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      
    </Text>
  </View>
  <TouchableOpacity
    onPress={[Function]}
    style={
      Object {
        "flex": 0.1,
        "paddingLeft": 5,
      }
    }
  >
    <Icon
      color="#616161"
      name="keyboard-arrow-up"
      size={25}
    />
  </TouchableOpacity>
</React.Fragment>
`;

exports[`SettingsToolScreen Tests rendering featureCard component featureCard function renders "Disabled" translation if isEnabled is set to False 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "backgroundColor": "#ffffffff",
      "borderTopColor": "#9e9e9e",
      "borderTopWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingBottom": 10,
      "paddingHorizontal": 10,
    }
  }
>
  <Text
    style={
      Object {
        "paddingTop": 5,
      }
    }
  >
    manager approval
  </Text>
  <Text
    style={
      Object {
        "paddingTop": 5,
      }
    }
  >
    [missing "en.GENERICS.DISABLED" translation]
  </Text>
</View>
`;

exports[`SettingsToolScreen Tests rendering featureCard component featureCard function renders "Enabled" translation if isEnabled is set to True 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "backgroundColor": "#ffffffff",
      "borderTopColor": "#9e9e9e",
      "borderTopWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingBottom": 10,
      "paddingHorizontal": 10,
    }
  }
>
  <Text
    style={
      Object {
        "paddingTop": 5,
      }
    }
  >
    manager approval
  </Text>
  <Text
    style={
      Object {
        "paddingTop": 5,
      }
    }
  >
    [missing "en.GENERICS.ENABLED" translation]
  </Text>
</View>
`;

exports[`SettingsToolScreen Tests rendering printerCard component printerCard functions renders "Not Assigned" translation if Printer is null 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffffff",
      "borderBottomColor": "#9e9e9e",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingBottom": 10,
      "paddingHorizontal": 10,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      Price Sign Printer
    </Text>
    <Text>
        
      [missing "en.GENERICS.NOT_ASSIGNED" translation]
    </Text>
  </View>
  <TouchableOpacity
    onPress={[MockFunction]}
    style={
      Object {
        "justifyContent": "flex-end",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#006DE5",
        }
      }
    >
      [missing "en.GENERICS.CHANGE" translation]
    </Text>
  </TouchableOpacity>
</View>
`;

exports[`SettingsToolScreen Tests rendering printerCard component printerCard functions renders Printer Name if Printer is not null 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffffff",
      "borderBottomColor": "#9e9e9e",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingBottom": 10,
      "paddingHorizontal": 10,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      Price Sign Printer
    </Text>
    <Text>
        
      Test Laser Printer
    </Text>
  </View>
  <TouchableOpacity
    onPress={[MockFunction]}
    style={
      Object {
        "justifyContent": "flex-end",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#006DE5",
        }
      }
    >
      [missing "en.GENERICS.CHANGE" translation]
    </Text>
  </TouchableOpacity>
</View>
`;

exports[`SettingsToolScreen Tests rendering the SettingsToolScreen component Test renders the SettingsToolScreen with Selected Printers 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffffff",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffffff",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingHorizontal": 10,
          "paddingVertical": 10,
        }
      }
    >
      <CollapsibleCard
        isOpened={true}
        title="Printers"
        toggleIsOpened={[MockFunction]}
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#ffffffff",
          "borderBottomColor": "#9e9e9e",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 10,
          "paddingHorizontal": 10,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "fontSize": 16,
            }
          }
        >
          [missing "en.PRINT.PRICE_SIGN_PRINTER" translation]
        </Text>
        <Text>
            
          Test Laser Printer
        </Text>
      </View>
      <TouchableOpacity
        onPress={[Function]}
        style={
          Object {
            "justifyContent": "flex-end",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#006DE5",
            }
          }
        >
          [missing "en.GENERICS.CHANGE" translation]
        </Text>
      </TouchableOpacity>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#ffffffff",
          "borderBottomColor": "#9e9e9e",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 10,
          "paddingHorizontal": 10,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "fontSize": 16,
            }
          }
        >
          [missing "en.PRINT.LOCATION_LABEL_PRINTER" translation]
        </Text>
        <Text>
            
          Test Portable Printer
        </Text>
      </View>
      <TouchableOpacity
        onPress={[Function]}
        style={
          Object {
            "justifyContent": "flex-end",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#006DE5",
            }
          }
        >
          [missing "en.GENERICS.CHANGE" translation]
        </Text>
      </TouchableOpacity>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#ffffffff",
          "borderBottomColor": "#9e9e9e",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 10,
          "paddingHorizontal": 10,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "fontSize": 16,
            }
          }
        >
          [missing "en.PRINT.PALLET_LABEL_PRINTER" translation]
        </Text>
        <Text>
            
          Test Laser Printer#2
        </Text>
      </View>
      <TouchableOpacity
        onPress={[Function]}
        style={
          Object {
            "justifyContent": "flex-end",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#006DE5",
            }
          }
        >
          [missing "en.GENERICS.CHANGE" translation]
        </Text>
      </TouchableOpacity>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffffff",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingHorizontal": 10,
          "paddingVertical": 10,
        }
      }
    >
      <CollapsibleCard
        isOpened={true}
        title="Features"
        toggleIsOpened={[MockFunction]}
      />
    </View>
    <ForwardRef
      ListFooterComponent={<View />}
      ListFooterComponentStyle={
        Object {
          "borderBottomWidth": 1,
        }
      }
      data={
        Array [
          Object {
            "key": "manager approval",
            "name": "[missing \\"en.APPROVAL.MANAGER_APPROVAL\\" translation]",
          },
          Object {
            "key": "location management",
            "name": "[missing \\"en.LOCATION.LOCATION_MANAGEMENT\\" translation]",
          },
          Object {
            "key": "on hands change",
            "name": "[missing \\"en.APPROVAL.OH_CHANGE\\" translation]",
          },
          Object {
            "key": "location management edit",
            "name": "[missing \\"en.LOCATION.LOCATION_MGMT_EDIT\\" translation]",
          },
          Object {
            "key": "location printing",
            "name": "[missing \\"en.PRINT.LOCATION_PRINTING\\" translation]",
          },
          Object {
            "key": "pallet management",
            "name": "[missing \\"en.LOCATION.PALLET_MANAGEMENT\\" translation]",
          },
          Object {
            "key": "picking",
            "name": "[missing \\"en.PICKING.PICKING\\" translation]",
          },
          Object {
            "key": "binning",
            "name": "[missing \\"en.BINNING.BINNING\\" translation]",
          },
        ]
      }
      keyExtractor={[Function]}
      renderItem={[Function]}
    />
  </View>
  <View
    style={
      Object {
        "marginHorizontal": 15,
        "marginVertical": 10,
      }
    }
  >
    <Button
      backgroundColor="#0071e9"
      disabled={false}
      height={45}
      onPress={[Function]}
      radius={5}
      testID="updateButton"
      title="[missing \\"en.GENERICS.UPDATE\\" translation]"
      titleColor="#ffffffff"
      titleFontSize={16}
      titleFontWeight="normal"
      type={1}
      width={-1}
    />
  </View>
</View>
`;

exports[`SettingsToolScreen Tests rendering the SettingsToolScreen component Test renders the SettingsToolScreen with the Feature List collapsed 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffffff",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffffff",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingHorizontal": 10,
          "paddingVertical": 10,
        }
      }
    >
      <CollapsibleCard
        isOpened={true}
        title="Printers"
        toggleIsOpened={[MockFunction]}
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#ffffffff",
          "borderBottomColor": "#9e9e9e",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 10,
          "paddingHorizontal": 10,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "fontSize": 16,
            }
          }
        >
          [missing "en.PRINT.PRICE_SIGN_PRINTER" translation]
        </Text>
        <Text>
            
          [missing "en.GENERICS.NOT_ASSIGNED" translation]
        </Text>
      </View>
      <TouchableOpacity
        onPress={[Function]}
        style={
          Object {
            "justifyContent": "flex-end",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#006DE5",
            }
          }
        >
          [missing "en.GENERICS.CHANGE" translation]
        </Text>
      </TouchableOpacity>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#ffffffff",
          "borderBottomColor": "#9e9e9e",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 10,
          "paddingHorizontal": 10,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "fontSize": 16,
            }
          }
        >
          [missing "en.PRINT.LOCATION_LABEL_PRINTER" translation]
        </Text>
        <Text>
            
          [missing "en.GENERICS.NOT_ASSIGNED" translation]
        </Text>
      </View>
      <TouchableOpacity
        onPress={[Function]}
        style={
          Object {
            "justifyContent": "flex-end",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#006DE5",
            }
          }
        >
          [missing "en.GENERICS.CHANGE" translation]
        </Text>
      </TouchableOpacity>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#ffffffff",
          "borderBottomColor": "#9e9e9e",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 10,
          "paddingHorizontal": 10,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "fontSize": 16,
            }
          }
        >
          [missing "en.PRINT.PALLET_LABEL_PRINTER" translation]
        </Text>
        <Text>
            
          [missing "en.GENERICS.NOT_ASSIGNED" translation]
        </Text>
      </View>
      <TouchableOpacity
        onPress={[Function]}
        style={
          Object {
            "justifyContent": "flex-end",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#006DE5",
            }
          }
        >
          [missing "en.GENERICS.CHANGE" translation]
        </Text>
      </TouchableOpacity>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffffff",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingHorizontal": 10,
          "paddingVertical": 10,
        }
      }
    >
      <CollapsibleCard
        isOpened={false}
        title="Features"
        toggleIsOpened={[MockFunction]}
      />
    </View>
  </View>
  <View
    style={
      Object {
        "marginHorizontal": 15,
        "marginVertical": 10,
      }
    }
  >
    <Button
      backgroundColor="#0071e9"
      disabled={false}
      height={45}
      onPress={[Function]}
      radius={5}
      testID="updateButton"
      title="[missing \\"en.GENERICS.UPDATE\\" translation]"
      titleColor="#ffffffff"
      titleFontSize={16}
      titleFontWeight="normal"
      type={1}
      width={-1}
    />
  </View>
</View>
`;

exports[`SettingsToolScreen Tests rendering the SettingsToolScreen component Test renders the SettingsToolScreen with the Printer List collapsed 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffffff",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffffff",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingHorizontal": 10,
          "paddingVertical": 10,
        }
      }
    >
      <CollapsibleCard
        isOpened={false}
        title="Printers"
        toggleIsOpened={[MockFunction]}
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffffff",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingHorizontal": 10,
          "paddingVertical": 10,
        }
      }
    >
      <CollapsibleCard
        isOpened={true}
        title="Features"
        toggleIsOpened={[MockFunction]}
      />
    </View>
    <ForwardRef
      ListFooterComponent={<View />}
      ListFooterComponentStyle={
        Object {
          "borderBottomWidth": 1,
        }
      }
      data={
        Array [
          Object {
            "key": "manager approval",
            "name": "[missing \\"en.APPROVAL.MANAGER_APPROVAL\\" translation]",
          },
          Object {
            "key": "location management",
            "name": "[missing \\"en.LOCATION.LOCATION_MANAGEMENT\\" translation]",
          },
          Object {
            "key": "on hands change",
            "name": "[missing \\"en.APPROVAL.OH_CHANGE\\" translation]",
          },
          Object {
            "key": "location management edit",
            "name": "[missing \\"en.LOCATION.LOCATION_MGMT_EDIT\\" translation]",
          },
          Object {
            "key": "location printing",
            "name": "[missing \\"en.PRINT.LOCATION_PRINTING\\" translation]",
          },
          Object {
            "key": "pallet management",
            "name": "[missing \\"en.LOCATION.PALLET_MANAGEMENT\\" translation]",
          },
          Object {
            "key": "picking",
            "name": "[missing \\"en.PICKING.PICKING\\" translation]",
          },
          Object {
            "key": "binning",
            "name": "[missing \\"en.BINNING.BINNING\\" translation]",
          },
        ]
      }
      keyExtractor={[Function]}
      renderItem={[Function]}
    />
  </View>
  <View
    style={
      Object {
        "marginHorizontal": 15,
        "marginVertical": 10,
      }
    }
  >
    <Button
      backgroundColor="#0071e9"
      disabled={false}
      height={45}
      onPress={[Function]}
      radius={5}
      testID="updateButton"
      title="[missing \\"en.GENERICS.UPDATE\\" translation]"
      titleColor="#ffffffff"
      titleFontSize={16}
      titleFontWeight="normal"
      type={1}
      width={-1}
    />
  </View>
</View>
`;

exports[`SettingsToolScreen Tests rendering the SettingsToolScreen component Test renders the default SettingsToolScreen  1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffffff",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffffff",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingHorizontal": 10,
          "paddingVertical": 10,
        }
      }
    >
      <CollapsibleCard
        isOpened={true}
        title="Printers"
        toggleIsOpened={[MockFunction]}
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#ffffffff",
          "borderBottomColor": "#9e9e9e",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 10,
          "paddingHorizontal": 10,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "fontSize": 16,
            }
          }
        >
          [missing "en.PRINT.PRICE_SIGN_PRINTER" translation]
        </Text>
        <Text>
            
          [missing "en.GENERICS.NOT_ASSIGNED" translation]
        </Text>
      </View>
      <TouchableOpacity
        onPress={[Function]}
        style={
          Object {
            "justifyContent": "flex-end",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#006DE5",
            }
          }
        >
          [missing "en.GENERICS.CHANGE" translation]
        </Text>
      </TouchableOpacity>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#ffffffff",
          "borderBottomColor": "#9e9e9e",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 10,
          "paddingHorizontal": 10,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "fontSize": 16,
            }
          }
        >
          [missing "en.PRINT.LOCATION_LABEL_PRINTER" translation]
        </Text>
        <Text>
            
          [missing "en.GENERICS.NOT_ASSIGNED" translation]
        </Text>
      </View>
      <TouchableOpacity
        onPress={[Function]}
        style={
          Object {
            "justifyContent": "flex-end",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#006DE5",
            }
          }
        >
          [missing "en.GENERICS.CHANGE" translation]
        </Text>
      </TouchableOpacity>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#ffffffff",
          "borderBottomColor": "#9e9e9e",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 10,
          "paddingHorizontal": 10,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "fontSize": 16,
            }
          }
        >
          [missing "en.PRINT.PALLET_LABEL_PRINTER" translation]
        </Text>
        <Text>
            
          [missing "en.GENERICS.NOT_ASSIGNED" translation]
        </Text>
      </View>
      <TouchableOpacity
        onPress={[Function]}
        style={
          Object {
            "justifyContent": "flex-end",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#006DE5",
            }
          }
        >
          [missing "en.GENERICS.CHANGE" translation]
        </Text>
      </TouchableOpacity>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffffff",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingHorizontal": 10,
          "paddingVertical": 10,
        }
      }
    >
      <CollapsibleCard
        isOpened={true}
        title="Features"
        toggleIsOpened={[MockFunction]}
      />
    </View>
    <ForwardRef
      ListFooterComponent={<View />}
      ListFooterComponentStyle={
        Object {
          "borderBottomWidth": 1,
        }
      }
      data={
        Array [
          Object {
            "key": "manager approval",
            "name": "[missing \\"en.APPROVAL.MANAGER_APPROVAL\\" translation]",
          },
          Object {
            "key": "location management",
            "name": "[missing \\"en.LOCATION.LOCATION_MANAGEMENT\\" translation]",
          },
          Object {
            "key": "on hands change",
            "name": "[missing \\"en.APPROVAL.OH_CHANGE\\" translation]",
          },
          Object {
            "key": "location management edit",
            "name": "[missing \\"en.LOCATION.LOCATION_MGMT_EDIT\\" translation]",
          },
          Object {
            "key": "location printing",
            "name": "[missing \\"en.PRINT.LOCATION_PRINTING\\" translation]",
          },
          Object {
            "key": "pallet management",
            "name": "[missing \\"en.LOCATION.PALLET_MANAGEMENT\\" translation]",
          },
          Object {
            "key": "picking",
            "name": "[missing \\"en.PICKING.PICKING\\" translation]",
          },
          Object {
            "key": "binning",
            "name": "[missing \\"en.BINNING.BINNING\\" translation]",
          },
        ]
      }
      keyExtractor={[Function]}
      renderItem={[Function]}
    />
  </View>
  <View
    style={
      Object {
        "marginHorizontal": 15,
        "marginVertical": 10,
      }
    }
  >
    <Button
      backgroundColor="#0071e9"
      disabled={false}
      height={45}
      onPress={[Function]}
      radius={5}
      testID="updateButton"
      title="[missing \\"en.GENERICS.UPDATE\\" translation]"
      titleColor="#ffffffff"
      titleFontSize={16}
      titleFontWeight="normal"
      type={1}
      width={-1}
    />
  </View>
</View>
`;
